src/test/java/uk/gov/hmcts/reform/unspec/bpmn/BpmnBaseTest.java

```diff
@@ -122,4 +130,76 @@ public ProcessDefinition getProcessDefinitionByMessage(String messageName) {
     public void completeTask(String taskId) {
         engine.getExternalTaskService().complete(taskId, WORKER_ID);
     }
+
+    /**
+     * Completes an external task with the given id and variables.
+     *
+     * @param taskId the id of the external task to complete.
+     */
+    public void completeTask(String taskId, VariableMap variables) {
+        engine.getExternalTaskService().complete(taskId, WORKER_ID, variables);
+    }
+
+    /**
+     * Get external task for topic name.
+     */
+    public ExternalTask getNextExternalTask(String topicName) {
+        assertThat(getTopics()).containsOnly(topicName);
+
+        List<ExternalTask> externalTasks = getExternalTasks();
+        assertThat(externalTasks).hasSize(1);
+
+        ExternalTask externalTask = externalTasks.get(0);
+        assertThat(externalTask.getTopicName()).isEqualTo(topicName);
+
+        return externalTask;
+    }
+
+    /**
+     * Completes the external task with topic name.
+     *
+     * @param externalTask the id of the external task to complete.
+     * @param topicName    is taskName.
+     * @param caseEvent    is input variable for external task.
+     */
+    public void completeExternalTask(ExternalTask externalTask, String topicName, String caseEvent, String activityId) {
+        completeExternalTask(externalTask, topicName, caseEvent, activityId, null);
+    }
+
+    /**
+     * Completes the external task with topic name and variables.
+     *
+     * @param externalTask the id of the external task to complete.
+     * @param topicName    is taskName.
+     * @param caseEvent    is input variable for external task.
+     */
+    public void completeExternalTask(
+        ExternalTask externalTask,
+        String topicName,
+        String caseEvent,
+        String activityId,
+        VariableMap variables
+    ) {
+        assertThat(externalTask.getTopicName()).isEqualTo(topicName);
+
+        List<LockedExternalTask> lockedProcessTask = fetchAndLockTask(topicName);
+
+        assertThat(lockedProcessTask).hasSize(1);
+
+        assertThat(lockedProcessTask.get(0).getVariables())
+            .containsEntry("caseEvent", caseEvent);
+
+        assertThat(lockedProcessTask.get(0).getActivityId()).isEqualTo(activityId);
+
+        if (Optional.ofNullable(variables).isPresent()) {
+            completeTask(lockedProcessTask.get(0).getId(), variables);
+        } else {
+            completeTask(lockedProcessTask.get(0).getId());
+        }
```

Comment:
I think we should avoid any kind of logic in tests code, should we change this method to `assertExternalTask` and just remove `completeTask` call?